akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -390,13 +390,25 @@ class HttpExt(config: Config)(implicit system: ActorSystem) extends akka.actor.E
     synchronized {
       _defaultClientHttpsContext match {
         case null ⇒
-          val ctx = HttpsContext(SSLContext.getDefault)
+          val ctx = createDefaultClientHttpsContext
           _defaultClientHttpsContext = ctx
           ctx
         case ctx ⇒ ctx
       }
     }
 
+  private def createDefaultClientHttpsContext: HttpsContext = {
+    val defaultCtx = SSLContext.getDefault
+
+    val params = new SSLParameters
+    if (!Java6Compat.setEndpointIdentificationAlgorithm(params, "https"))
+      throw new ReadTheDocumentationException(
```

Comment:
I don't mind the `ReadTheDocumentationException`, sounds good (hey, it's not RTFM ;-))